### PR TITLE
support display app name for android

### PR DESCRIPTION
**Pull request by magaofei**
### Describe the Pull Request

```
By default, the file name and package name are displayed, and the name of the application is not displayed.
```

### Checklist for PR

- [x] Run MobSF unit tests (http://your-mobsf-@ip:8000/tests/ or python3 manage.py test).
- [x] Tested Working on Linux, Mac, Windows, and Docker
- [x] Make sure build is passing on your PR. [Build Status](https://travis-ci.com/MobSF/Mobile-Security-Framework-MobSF/pull_requests)
- [x] Coding style (indentation, etc)

### Additional Comments (if any)

```
![image](https://user-images.githubusercontent.com/3363787/58887084-8791fd00-8717-11e9-961d-d1b7d2255423.png) 
```

### CODE REVIEW

File: StaticAnalyzer/views/android/static_analyzer.py [hunk 1]
```diff
@@ -508,3 +524,24 @@ def gen_downloads(app_dir, md5, icon_path=''):
...
+
+def get_app_name(app_path, app_dir, tools_dir, is_apk):
+    """ get app name """
+    
+    data = ""
+    if is_apk:
+        output_dir = os.path.join(app_dir, "apktool_out")
+        strings_file = os.path.join(output_dir, "res/values/strings.xml")
+    else:
+        strings_file = os.path.join(app_dir, 'app/src/main/res/values/strings.xml')
```
Comment (ajinabraham) [inline StaticAnalyzer/views/android/static_analyzer.py]: What if it's an old Eclipse style android project?
`app/src/main/res/values/` directory exists?

Comment (magaofei) [reply StaticAnalyzer/views/android/static_analyzer.py]: I don't know, I don't have many test samples.

**Comment by ajinabraham** [comment]
@magaofei Can you zip this: https://github.com/Apress/android-apps-w-eclipse/tree/master/9781430244349/Chapter%206/MoviePlayer and try. We should not break eclipse support.

**Comment by magaofei** [comment]
> @magaofei Looks good to me. Please see the comments.
> Also real app name is missing in PDF templates

```
Zipped Source Code Instruction
Mobile Security Framework supports iOS and Android (eclipse and Android Studio) project files.

Eclipse Project: The file AndroidManifest.xml and the directory 'src' must exist in the root directory of ZIPPED Source files for eclipse projects.

Android Studio Project: AndroidManifest.xml must be located at 'app/src/main/AndroidManifest.xml' and the directory 'app/src/main/java/' must exist for Android Studio projects.

iOS Project: For iOS projects, the .xcodeproj file must exist in the root directory of the ZIPPED Source files.
```

**Comment by ajinabraham** [comment]
You might haven't zipped the root directory. Try this 
[Archive.zip](https://github.com/MobSF/Mobile-Security-Framework-MobSF/files/3307112/Archive.zip)

**Comment by magaofei** [comment]
> You might haven't zipped the root directory. Try this
> [Archive.zip](https://github.com/MobSF/Mobile-Security-Framework-MobSF/files/3307112/Archive.zip)

already change

File: StaticAnalyzer/views/android/static_analyzer.py [hunk 2]
```diff
@@ -508,3 +524,27 @@ def gen_downloads(app_dir, md5, icon_path=''):
...
+
+def get_app_name(app_path, app_dir, tools_dir, is_apk):
+    """ get app name """
+    
+    data = ""
+    if is_apk:
+        output_dir = os.path.join(app_dir, "apktool_out")
+        strings_file = os.path.join(output_dir, "res/values/strings.xml")
+    else:
+        if os.path.exists('app/src/main/res/values/strings.xml'):
```
Comment (ajinabraham) [inline StaticAnalyzer/views/android/static_analyzer.py]: This is not an absolute path

File: StaticAnalyzer/views/android/static_analyzer.py [hunk 3]
```diff
@@ -508,3 +524,27 @@ def gen_downloads(app_dir, md5, icon_path=''):
...
+            strings_file = os.path.join(app_dir, 'app/src/main/res/values/strings.xml')
+        else:
+            strings_file = os.path.join(app_dir, 'res/values/strings.xml')
```
Comment (ajinabraham) [inline StaticAnalyzer/views/android/static_analyzer.py]: What if the path doesn't exist? `else` may not be a good idea.

File: StaticAnalyzer/views/android/static_analyzer.py (hunk 2 above)
Comment (ajinabraham) [reply StaticAnalyzer/views/android/static_analyzer.py]: You missed `app_dir` I guess.

File: StaticAnalyzer/views/android/static_analyzer.py (hunk 3 above)
Comment (ajinabraham) [reply StaticAnalyzer/views/android/static_analyzer.py]: The pseudo logic may be like
if apk and the strings.xml exists 
 -> set path
else if studio projects strings.xml path exist
  -> set path
else if eclipse project strings.xml path exist
  -> set path
else 
   return an empty string # couldn't find the real app name, also doesn't break the further processing.

File: StaticAnalyzer/views/android/static_analyzer.py [hunk 4]
```diff
@@ -508,3 +524,27 @@ def gen_downloads(app_dir, md5, icon_path=''):
...
+
+    with open(strings_file, 'r', encoding='utf-8') as f:
```
Comment (ajinabraham) [inline StaticAnalyzer/views/android/static_analyzer.py]: See the comment above. If the above path doesn't exist. it will break the processing.

File: templates/pdf/android_source_analysis_pdf.html [hunk 5]
```diff
@@ -66,6 +66,10 @@ <h3 class="panel-title"><i class="fa fa-clock-o fa-fw"></i> <span class="glyphic
                             </div>
                             <div class="panel-body">
                                 <div class="list-group">
+                                    <p>                                       
+                                        <span class="label label-warning">Name</span>
```
Comment (ajinabraham) [inline templates/pdf/android_source_analysis_pdf.html]: Why `warning` ? It should follow the same UI style as other components in the same block.

File: StaticAnalyzer/views/android/static_analyzer.py [hunk 6]
```diff
@@ -508,3 +524,32 @@ def gen_downloads(app_dir, md5, icon_path=''):
...
+        strings_file = os.path.join(output_dir, "res/values/strings.xml")
+    else:
+        strings_path = os.path.join(app_dir, 'app/src/main/res/values/strings.xml')
+        eclipse_path = os.path.join(app_dir, 'res/values/strings.xml')
+        if os.path.exists(strings_path):
+            strings_file = strings_path
+        elif os.path.exists(eclipse_path):
+            strings_file = eclipse_path
+        else:
+            logger.warning('''can't find android name''')
```
Comment (ajinabraham) [inline StaticAnalyzer/views/android/static_analyzer.py]: Can you reword this to `Cannot find app name`.

File: StaticAnalyzer/views/android/static_analyzer.py [hunk 7]
```diff
@@ -508,3 +524,32 @@ def gen_downloads(app_dir, md5, icon_path=''):
...
         PrintException("Generating Downloads")
+
+
+def get_app_name(app_path, app_dir, tools_dir, is_apk):
+    """ get app name """
+    
+    data = ""
+    if is_apk:
+        output_dir = os.path.join(app_dir, "apktool_out")
+        strings_file = os.path.join(output_dir, "res/values/strings.xml")
```
Comment (ajinabraham) [inline StaticAnalyzer/views/android/static_analyzer.py]: Are you sure for all versions of APK
`(output_dir, "res/values/strings.xml")` this path exists?

If not `os.path.exists` needs to be done before attempting to read strings.xml in all case.

Comment (magaofei) [reply StaticAnalyzer/views/android/static_analyzer.py]: I added a judgment to the back that did not find the string_file file.
